File: Python/Carro.py
class carro:
        def __init__(self,marca,modelo,ano,ligado):
                self.marca = marca
                self.modelo = modelo
                self.ano = ano
                self.ligado = False
        
        def ligar(self):
                self.ligado = True
                print(f'Carro ligado!!!')
       
        
        def desligar(self):
            self.ligado = False
            print(f'Carro desligado!!!')

        
        def mostrarInfo(self):
            print(f'marca: {self.marca}')
            print(f'modelo: {self.modelo}')
            print(f'ano: {self.ano}')
        
    
    
    
def main():
    carro1= carro('ford','fiesta','2020',False)
    carro1.mostrarInfo()
    print('**********')

    
    carro1.ligar()
    carro1.mostrarInfo()
    print('*********')
    carro1.desligar()

    
    carro1.mostrarInfo()
if __name__=="__main__":
        main()

File: IDE/encapsulamento/src/main/kotlin/Carro.kt
class Carro(val modelo: String, private var ligado: Boolean, private var velocidade: Double) {
    fun ligar() {
        ligado = true
    }

    fun desligar() {
        if (velocidade == 0.0) {
            ligado = false
        }
    }

    fun freiar(): Double {
        if (ligado) {
            velocidade -= 3.0
            if (velocidade < 3.0) {
                velocidade = 0.0
                return velocidade
            }
            return velocidade
        } else {
            return velocidade
        }
    }
    fun acelerar(): Double {
        return if (ligado) {
            if (velocidade == 0.0) {
                velocidade = 10.00
                velocidade
            } else {
                velocidade += velocidade * 0.25
                if (velocidade >= 180.00) {
                    velocidade = 180.00
                }
                velocidade
            }
        } else {
            velocidade
        }
    }


    fun getVelocidade(): Double{
        return velocidade
    }
}